aws_utils/s3/s3_utils.py

```diff
@@ -487,3 +488,75 @@ def rename_keys_on_s3(bucket_name, bucket_region, prefix_root, prefix_modificati
                 rename_s3_key(boto3_client, bucket_name, current_prefix, new_prefix_name)
             except (ClientError, AttributeError) as e:
                 logger.error('Unable to rename key prefix {}, {}'.format(current_prefix, e[0]))
+
+
+def get_s3_keys_by_regex(s3_bucket, s3_directory, pattern):
```

Comment:
Standard convertion is to use get in the name for a single item but fetch for multiple, maybe use fetch here?